Jenkinsfile

```diff
@@ -113,34 +166,26 @@ pipeline {
                                 # todo: include ip_vs in base image
                                 sudo modprobe ip_vs
                 
-                                GITCOMMIT=$(git rev-parse --short HEAD)
-                                docker build --force-rm --build-arg APT_MIRROR -t docker:$GITCOMMIT .
+                                docker build --force-rm --build-arg APT_MIRROR -t docker:${GIT_COMMIT} .
                                 '''
                             }
                         }
                         stage("Run tests") {
                             steps {
                                 sh '''
-                                GITCOMMIT=$(git rev-parse --short HEAD)
-        
                                 docker run --rm -t --privileged \
                                   -v "$WORKSPACE/bundles:/go/src/github.com/docker/docker/bundles" \
                                   -v "$WORKSPACE/.git:/go/src/github.com/docker/docker/.git" \
                                   --name docker-pr$BUILD_NUMBER \
-                                  -e DOCKER_GITCOMMIT=${GITCOMMIT} \
-                                  -e DOCKER_GRAPHDRIVER=overlay2 \
-                                  -e GIT_SHA1=${GIT_COMMIT} \
-                                  docker:$GITCOMMIT \
-                                  hack/ci/janky
-                                '''
-                            }
-                        }
-                        stage("Build e2e image") {
-                            steps {
-                                sh '''
-                                GITCOMMIT=$(git rev-parse --short HEAD)
-                                echo "Building e2e image"
-                                docker build --build-arg DOCKER_GITCOMMIT=$GITCOMMIT -t moby-e2e-test -f Dockerfile.e2e .
+                                  -e DOCKER_GITCOMMIT=${GIT_COMMIT} \
+                                  -e DOCKER_GRAPHDRIVER \
+                                  docker:${GIT_COMMIT} \
+                                  hack/make.sh \
+                                    binary-daemon \
+                                    dynbinary-daemon \
+                                    test-integration-flaky \
+                                    test-integration \
```

Comment:
Should probably split this up into separate stages as well, but here I was also cautious because we want to archive artifacts